satellite/certdb/certdb_test.go

```diff
@@ -31,75 +32,63 @@ func testDatabase(ctx context.Context, t *testing.T, upldb certdb.DB) {
 	{ //uplink testing variables
 		upID, err := testidentity.NewTestIdentity(ctx)
 		require.NoError(t, err)
-		upIDpubbytes, err := x509.MarshalPKIXPublicKey(upID.Leaf.PublicKey)
+		pi := upID.PeerIdentity()
+		upIDpubbytes, err := pkcrypto.PublicKeyToPEM(pi.CA.PublicKey)
 		require.NoError(t, err)
 
 		{ // New entry
-			err := upldb.SavePublicKey(ctx, upID.ID, upID.Leaf.PublicKey)
+			err := upldb.SavePublicKey(ctx, upID.ID, pi)
 			assert.NoError(t, err)
 		}
 
 		{ // Get the corresponding Public key for the serialnum
-			pubkey, err := upldb.GetPublicKey(ctx, upID.ID)
+			uplpi, err := upldb.GetPublicKey(ctx, upID.ID)
 			assert.NoError(t, err)
-			pubbytes, err := x509.MarshalPKIXPublicKey(pubkey)
-			assert.NoError(t, err)
-			assert.EqualValues(t, upIDpubbytes, pubbytes)
-		}
-
-		{ // Get the corresponding Public key for the serialnum
-			pubkey, err := upldb.GetPublicKeys(ctx, upID.ID)
-			assert.NoError(t, err)
-			pubbytes, err := x509.MarshalPKIXPublicKey(pubkey[0])
+			pubbytes, err := pkcrypto.PublicKeyToPEM(uplpi.CA.PublicKey)
 			assert.NoError(t, err)
 			assert.EqualValues(t, upIDpubbytes, pubbytes)
 		}
 	}
 
 	{ //storagenode testing variables
-		sn1ID, err := testidentity.NewTestIdentity(ctx)
-		require.NoError(t, err)
-		sn1IDpubbytes, err := x509.MarshalPKIXPublicKey(sn1ID.Leaf.PublicKey)
+		sn1FI, err := testidentity.NewTestIdentity(ctx)
 		require.NoError(t, err)
+		sn1PI := sn1FI.PeerIdentity()
 
 		{ // New entry
-			err := upldb.SavePublicKey(ctx, sn1ID.ID, sn1ID.Leaf.PublicKey)
+			err := upldb.SavePublicKey(ctx, sn1PI.ID, sn1PI)
 			assert.NoError(t, err)
 		}
-
-		sn2ID, err := testidentity.NewTestIdentity(ctx)
+		sn2FI, err := testidentity.NewTestIdentity(ctx)
 		require.NoError(t, err)
-		sn2IDpubbytes, err := x509.MarshalPKIXPublicKey(sn2ID.Leaf.PublicKey)
+		sn2PI := sn2FI.PeerIdentity()
+		sn2PIpubbytes, err := pkcrypto.PublicKeyToPEM(sn2PI.CA.PublicKey)
 		require.NoError(t, err)
 
-		{ // add another key for the same storagenode ID
-			err := upldb.SavePublicKey(ctx, sn1ID.ID, sn2ID.Leaf.PublicKey)
-			assert.NoError(t, err)
-		}
-
-		{ // add another key for the same storagenode ID, this the latest key
-			err := upldb.SavePublicKey(ctx, sn1ID.ID, sn2ID.Leaf.PublicKey)
-			assert.NoError(t, err)
+		{ // adding two different pubkeys for same storagnode
+			{ // add a key for to storagenode ID
+				err := upldb.SavePublicKey(ctx, sn2PI.ID, sn1PI)
+				assert.NoError(t, err)
+			}
+			time.Sleep(5)
```

Comment:
Remove the sleep.

Reply:
will do .. I added to create a little time different in two entries with same nodeID but different peer identity. But again might change based on my above comment, if only one nodeID exists or can there be more than one entry with same nodeIDs but different peer identities...